tests/production_pause_vote_timeout.py

```diff
@@ -0,0 +1,197 @@
+    # Verify producercNode paused
+    assert paused, "producercNode still producing after finalizercNode was shutdown"
+    # Verify node0 and node1 still producing
+    assert node0.waitForHeadToAdvance(), "node0 paused after finalizercNode was shutdown"
+    assert node1.waitForHeadToAdvance(), "node1 paused after finalizercNode was shutdown"
+
+    Print("Restart finalizercNode")
+    finalizercNode.relaunch()
+
+    Print("Verify LIB advances after restart of finalizercNode")
+    assert node0.waitForLibToAdvance(), "node0 did not advance LIB"
+    assert node1.waitForLibToAdvance(), "node1 did not advance LIB"
+    assert producercNode.waitForLibToAdvance(), "producercNode did not advance LIB"
+
+    ####################### test 2 ######################
+
+    Print("Shutdown centerNode")
+    centerNode.kill(signal.SIGTERM)
+    assert not centerNode.verifyAlive(), "centerNode did not shutdown"
+
+    # wait some time for producercNode paused
+    paused = False
+    for i in range(0, 15):
+        time.sleep(1)
+        paused = not producercNode.waitForHeadToAdvance(timeout=1)
+        if paused:
+            Print(f'paused after {i} seconds after centerNode was shutdown')
+            break;
+    # Verify producercNode paused
+    assert paused, "producercNode still producing after centerNode was shutdown"
+    # Verify node0 and node1 still producing
+    assert node0.waitForHeadToAdvance(), "node0 paused after centerNode was shutdown"
+    assert node1.waitForHeadToAdvance(), "node1 paused after centerNode was shutdown"
+
+    Print("Restart centerNode")
+    centerNode.relaunch()
+
+    Print("Verify LIB advances after restart")
+    assert node0.waitForLibToAdvance(), "node0 did not advance LIB"
+    assert node1.waitForLibToAdvance(), "node1 did not advance LIB"
+    assert producercNode.waitForLibToAdvance(), "producercNode did not advance LIB"
+
+    ####################### test 3 ######################
+
+    Print("Shutdown producercNode")
+    producercNode.kill(signal.SIGTERM)
+    assert not producercNode.verifyAlive(), "producercNode did not shutdown"
+
+    # disable production-pause-vote-timeout
+    Print("Relaunch producercNode with --production-pause-vote-timeout-ms 0")
+    addSwapFlags={"--production-pause-vote-timeout-ms": "0"}
+    producercNode.relaunch(addSwapFlags=addSwapFlags)
+
+    Print("Shutdown finalizercNode")
+    finalizercNode.kill(signal.SIGTERM)
+    assert not finalizercNode.verifyAlive(), "finalizercNode did not shutdown"
+
+    # wait some time to make sure it is not paused
+    paused = False
+    for i in range(0, 10):
+        time.sleep(1)
+        paused = not producercNode.waitForHeadToAdvance()
+        if paused:
+            Print(f'paused after {i} seconds after finalizercNode was shutdown')
+            break;
+    # Verify producercNode still producing
+    assert not paused, "producercNode (--production-pause-vote-timeout-ms 0) paused after finalizercNode was shutdown"
```

Comment:
This and the for loop can be simplified to just:
```
assert producercNode.waitForHeadToAdvance()
```
Since it will shortcut out as soon as head advances.